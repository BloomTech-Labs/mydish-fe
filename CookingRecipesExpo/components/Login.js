import React, {useState, useEffect} from 'react';
import {
  StyleSheet,
  View,
  Text,
  TextInput,
  TouchableOpacity,
  ScrollView,
  AsyncStorage
} from 'react-native';
import axios from 'axios'
// import AsyncStorage from '@react-native-community/async-storage'
// import MyCookBook from "./MyCookBook"
// import AxiosWithAuth from './AxiosWithAuth.js'
import styles from '../styles/loginStyles.js'
import MyCookBook from "./MyCookBook.js"



const Login = props => {
  const [login, SetLogin] = useState({username: '', password: ''})
  const [toke, setTok] = useState()

  const signInAsync = async (tok) => {
    await AsyncStorage.setItem('userToken', tok);
    props.navigation.navigate('App');
    const token = await AsyncStorage.getItem('userToken');
  };

//  useEffect(()=>{
//    <MyCookBook name={login.username}/>
//  },[login.username])

  const onPress = () => {
    axios.post('https://recipeshare-development.herokuapp.com/cooks/login', login)
    .then(res => {signInAsync(res.data.token),  console.log(res.data)})
      // console.log('response from login axios post', res.data.token)
    .catch(err => setTok(err))
    } 

    //console.log("tiktok",toke)
  return (
    <ScrollView>
    <View style={styles.signUp}>
      <TouchableOpacity
          onPress={() => props.navigation.navigate('Home')}
          >
            <Text style={styles.exitButton}>x</Text>
          </TouchableOpacity>
          <Text style={styles.title}>RecipeShare</Text>
          <Text style={styles.explanationText}>Sign in or create a new account to save and edit your favorite recipes.</Text>
          <Text style={styles.loginText}>Log In</Text>
          <Text style={styles.emailText}>Username</Text>
          <TextInput
           style={styles.inputFeilds}
           name="username"
           value={login.username}
           onChangeText={event => SetLogin({...login, username:event})}/>
           <Text style={styles.passwordText}>Password</Text>
          <TextInput
           style={styles.inputFeilds}
           name="password"
           value={login.password}
           onChangeText={event => SetLogin({...login, password:event})}
           secureTextEntry={true}/>
           {toke!=null && <Text style={{color:"red", marginLeft:100}}>Incorrect Username or Password</Text>}
           <TouchableOpacity
           onPress={() => props.navigation.navigate('Signup')}>
           <Text style={styles.createAccountButton}>Create an Account</Text>
           </TouchableOpacity>
           <View style={{flexDirection: 'row-reverse', marginRight: 16}}>
           <TouchableOpacity
             onPress={onPress}
             style={styles.loginButton}
           >
             <Text style={styles.loginButtonText}>Login</Text>
             {/* <MyCookBook props={toke} style={{display: 'none'}}/> */}
             {/* <AxiosWithAuth token={toke}/> */}
           </TouchableOpacity>
           </View>
        </View>
       </ScrollView>
       
      );
    }
    


export default Login;