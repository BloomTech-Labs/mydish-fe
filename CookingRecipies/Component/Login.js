import React, {useState} from 'react';
import {
  StyleSheet,
  View,
  Text,
  TextInput,
  TouchableOpacity,
} from 'react-native';
import axios from 'axios'
import AsyncStorage from '@react-native-community/async-storage'

import styles from '../styles/loginStyles.js'



const Login = props => {
const [login, SetLogin] = useState({username: '', password: ''})

const signInAsync = async () => {
  await AsyncStorage.setItem('userToken', 'abc');
  props.navigation.navigate('App');
  const token = await AsyncStorage.getItem('userToken')
  console.log(token)
};

console.log(login)

const onPress = () => {
  signInAsync()
 console.log("axios call goes here")
  // axios.put('', login)
  // .then(res => console.log('response from login axios post', res))
  // .catch(err => console.log('error from login axios post',err))
}

  return (
    <View style={styles.signUp}>
          <Text style={styles.title}>RecipeShare</Text>
          <Text style={styles.explanationText}>Sign in or create a new account to save and edit your favorite recipes.</Text>
          <Text style={styles.loginText}>Log In</Text>
          <Text style={styles.emailText}>Email</Text>
          <TextInput
           style={styles.inputFeilds}
           name="username"
           value={login.username}
           onChangeText={event => SetLogin({...login, username:event})}/>
           <Text style={styles.passwordText}>Password</Text>
          <TextInput
           style={styles.inputFeilds}
           name="password"
           value={login.password}
           onChangeText={event => SetLogin({...login, password:event})}/>
           <TouchableOpacity
           onPress={() => props.navigation.navigate('SignUp')}>
           <Text style={styles.createAccountButton}>Create an Account</Text>
           </TouchableOpacity>
           <TouchableOpacity
             onPress={onPress}
             style={styles.loginButton}
           >
             <Text style={styles.loginButtonText}>Login</Text>
           </TouchableOpacity>
        </View>
       
      );
    }
    


export default Login;