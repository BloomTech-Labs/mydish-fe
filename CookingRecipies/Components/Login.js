import React, {useState} from 'react';
import {
  StyleSheet,
  View,
  Text,
  TextInput,
  TouchableOpacity,
} from 'react-native';
import axios from 'axios'
import AsyncStorage from '@react-native-community/async-storage'

import styles from '../styles/loginStyles.js'



const Login = props => {
const [login, SetLogin] = useState({username: '', password: ''})

const signInAsync = async (tok) => {
  await AsyncStorage.setItem('userToken', tok);
  props.navigation.navigate('App');
  const token = await AsyncStorage.getItem('userToken')
  console.log(token)
};

console.log(login)

const onPress = () => {
  
 console.log("axios call goes here")
  axios.post('https://recipeshare-development.herokuapp.com/cooks/login', login)
  .then(res => signInAsync(res.data.token)
    // console.log('response from login axios post', res.data.token)
          )
  .catch(err => console.log('error from login axios post',err))
}

  return (
    <View style={styles.signUp}>
      <TouchableOpacity
          onPress={() => props.navigation.navigate('Home')}
          >
            <Text style={styles.exitButton}>x</Text>
          </TouchableOpacity>
          <Text style={styles.title}>RecipeShare</Text>
          <Text style={styles.explanationText}>Sign in or create a new account to save and edit your favorite recipes.</Text>
          <Text style={styles.loginText}>Log In</Text>
          <Text style={styles.emailText}>Email</Text>
          <TextInput
           style={styles.inputFeilds}
           name="username"
           value={login.username}
           onChangeText={event => SetLogin({...login, username:event})}/>
           <Text style={styles.passwordText}>Password</Text>
          <TextInput
           style={styles.inputFeilds}
           name="password"
           value={login.password}
           onChangeText={event => SetLogin({...login, password:event})}/>
           <TouchableOpacity
           onPress={() => props.navigation.navigate('SignUp')}>
           <Text style={styles.createAccountButton}>Create an Account</Text>
           </TouchableOpacity>
           <TouchableOpacity
             onPress={onPress}
             style={styles.loginButton}
           >
             <Text style={styles.loginButtonText}>Login</Text>
           </TouchableOpacity>
        </View>
       
      );
    }
    


export default Login;