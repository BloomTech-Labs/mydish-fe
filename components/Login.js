import React, { useState, useRef, useEffect } from 'react';
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  SafeAreaView,
  Alert,
  KeyboardAvoidingView,
  ScrollView,
  ImageBackground,
} from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { clearError, loginUser } from '../store/auth/authActions';
import styles from '../styles/authPageStyles.js';
import backgroundImg from '../assets/auth-page-background.jpg';
import AuthButton from './AuthButton';
import { logoHeaderPlain } from './header/navigationHeader';

//Analytics
import { Analytics, Event } from 'expo-analytics';

const analytics = new Analytics('UA-160806654-1');

const Login = ({ navigation }) => {
  const [credentials, setLogin] = useState({ username: '', password: '' });
  const errorMsg = useSelector((state) => state.auth.error);
  const dispatch = useDispatch();
  const usernameInput = useRef(null);
  const passwordInput = useRef(null);

  useEffect(() => {
    if (errorMsg != null) {
      emptyFieldsAlert();
    }
  }, [errorMsg]);

  const _loginUser = async () => {
    const success = await dispatch(loginUser(credentials));

    if (success) {
      navigation.navigate('App');
    }
    analytics
      .event(new Event('Login', 'User logged in'))
      .then(() => console.log('User logged in'))
      .catch((e) => console.log(e.message));
  };

  const emptyFieldsAlert = () => {
    return Alert.alert(
      'Oops!',
      'Please provide a valid username and password.',
      [{ title: 'Okay' }]
    );
  };

  return (
    <SafeAreaView>
      <ImageBackground source={backgroundImg} style={styles.backgroundImg}>
        <KeyboardAvoidingView behavior="height">
          <ScrollView>
            <View style={styles.container}>
              <View style={styles.contentContainer}>
                <Text style={styles.inputLabelText}>Username</Text>
                <TextInput
                  ref={usernameInput}
                  style={styles.inputFields}
                  name="username"
                  testID="username"
                  value={credentials.username}
                  returnKeyType="next"
                  onSubmitEditing={() => passwordInput.current.focus()}
                  onChangeText={(event) =>
                    setLogin({
                      ...credentials,
                      username: event,
                    })
                  }
                />
                <Text style={styles.inputLabelText}>Password</Text>
                <TextInput
                  ref={passwordInput}
                  style={styles.inputFields}
                  name="password"
                  testID="password"
                  value={credentials.password}
                  returnKeyType="done"
                  onChangeText={(event) =>
                    setLogin({
                      ...credentials,
                      password: event,
                    })
                  }
                  secureTextEntry={true}
                  onSubmitEditing={_loginUser}
                />

                <View style={styles.promptContainer}>
                  <Text style={styles.questionPrompt}>
                    Don't have an account?{' '}
                  </Text>
                  <TouchableOpacity
                    onPress={() => {
                      dispatch(clearError());
                      navigation.navigate('Signup');
                    }}
                  >
                    <Text style={styles.switchAuthPageLink}>Sign Up</Text>
                  </TouchableOpacity>
                </View>

                <View
                  style={{
                    flexDirection: 'row-reverse',
                    marginRight: 16,
                  }}
                >
                  <AuthButton
                    parent="login"
                    credentials={credentials}
                    loginUser={_loginUser}
                  />
                </View>
              </View>
            </View>
          </ScrollView>
        </KeyboardAvoidingView>
      </ImageBackground>
    </SafeAreaView>
  );
};
Login.navigationOptions = logoHeaderPlain;

export default Login;
