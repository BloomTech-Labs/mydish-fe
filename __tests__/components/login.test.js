import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';
import Login from '../../components/Login';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

test('login snapshot', () => {
  // The login component only needs the error portion of the store,
  //     so we create a simpler store doing just that:
  const store = createStore(() => ({ auth: { error: null } }));

  // Then we create our smaller Login component with the provider
  const renderTree = (
    <Provider store={store}>
      <Login />
    </Provider>
  );

  const loginSnap = renderer.create(renderTree).toJSON();
  expect(loginSnap).toMatchSnapshot();
});

describe('<Login/>', () => {
  it('Username and Password accepts input', () => {
    const store = createStore(() => ({ auth: { error: null } }));

    const renderTree = (
      <Provider store={store}>
        <Login />
      </Provider>
    );
    const { getByTestId } = render(renderTree);
    const UserInput = getByTestId('username');
    const PassInput = getByTestId('password');
    fireEvent.changeText(UserInput, 'Amir');
    expect(UserInput.props.value).toEqual('Amir');
    fireEvent.changeText(PassInput, 'password');
    expect(PassInput.props.value).toEqual('password');
  });
});
