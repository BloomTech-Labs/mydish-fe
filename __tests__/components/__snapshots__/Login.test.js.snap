// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Image
    source={1}
    style={
      Object {
        "maxHeight": "90%",
        "maxWidth": "80%",
        "position": "absolute",
        "resizeMode": "cover",
        "zIndex": -10000,
      }
    }
  />
  <View
    onLayout={[Function]}
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "height": "100%",
              "paddingHorizontal": 50,
              "paddingTop": 70,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "height": "65%",
                "justifyContent": "space-between",
                "paddingVertical": 70,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "marginBottom": 5,
                }
              }
            >
              Username
            </Text>
            <TextInput
              allowFontScaling={true}
              name="username"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              rejectResponderTermination={true}
              returnKeyType="next"
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#D4D4D4",
                  "borderRadius": 6,
                  "borderWidth": 1,
                  "color": "#959595",
                  "height": 40,
                  "marginBottom": 25,
                  "minHeight": "5%",
                  "paddingLeft": 10,
                }
              }
              testID="username"
              underlineColorAndroid="transparent"
              value=""
            />
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "marginBottom": 5,
                }
              }
            >
              Password
            </Text>
            <TextInput
              allowFontScaling={true}
              name="password"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              rejectResponderTermination={true}
              returnKeyType="done"
              secureTextEntry={true}
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#D4D4D4",
                  "borderRadius": 6,
                  "borderWidth": 1,
                  "color": "#959595",
                  "height": 40,
                  "marginBottom": 25,
                  "minHeight": "5%",
                  "paddingLeft": 10,
                }
              }
              testID="password"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "marginTop": -20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                  }
                }
              >
                Don't have an account?
                 
              </Text>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#D2291F",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "textDecorationLine": "underline",
                    }
                  }
                >
                  Sign Up
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row-reverse",
                  "marginRight": 16,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#D2291F",
                    "borderRadius": 5,
                    "height": 40,
                    "justifyContent": "center",
                    "marginTop": 30,
                    "opacity": 1,
                    "paddingVertical": 10,
                    "width": 150,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    }
                  }
                >
                  Login
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
